lib/_http_client.js

```diff
@@ -858,9 +858,11 @@ function listenSocketTimeout(req) {
 }
 
 ClientRequest.prototype.onSocket = function onSocket(socket, err) {
-  // TODO(ronag): Between here and onSocketNT the socket
-  // has no 'error' handler.
-  process.nextTick(onSocketNT, this, socket, err);
+  if (err) {
+    process.nextTick(() => this.emit('error', err));
```

Comment:
```suggestion
    process.nextTick(() => this.emit('error', err));
    return;
```

I think you forgot a `return`?